### PR TITLE
Add transaction book

### LINKED ISSUE #54: Implement Recordbook class
Stores all transactions.

### PR DESCRIPTION
**Pull request by Bombbird2001**
A lot of stuff changed, main differences:
- Refactor UniquePersonList, UniqueTransactionList to UniqueEntryList\<E extends Entry\>
- Person and Transaction classes implement the Entry interface which provides isSameEntry method (same as the old isSamePerson method in Person class)
- Rename exceptions that used "Person" in their names to use "Entry" instead
- Rename TransactionLog to TransactionBook, more or less the same as AddressBook class but for transactions; other classes with TransactionLog in their names also changed to use TransactionBook
- Integrate loading of transaction CSV (at app start), saving of transaction (on command execution)
- Update TransactionTablePanel to display the new Transaction properties
- Add TransactionBook related tests

Things still needed:
- More tests for Transaction class
- Transaction class still needs a Date
- Transaction parsing class should parse for the new fields (currently using placeholder)

Resolves #54

### CODE REVIEW

File: src/main/java/transact/logic/commands/AddTransactionCommand.java [hunk 1]
```diff
@@ -52,11 +53,11 @@ public CommandResult execute(Model model) throws CommandException {
...
 
         if (model.hasTransaction(toAdd)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            throw new CommandException(MESSAGE_DUPLICATE_TRANSACTION);
         }
 
         model.addTransaction(toAdd);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)));
+        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)), MainWindow.TabWindow.TRANSACTIONS);
     }
```
Comment (cxo05) [inline src/main/java/transact/logic/commands/AddTransactionCommand.java:61]:
```suggestion
        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)), TabWindow.TRANSACTIONS);
```

File: src/main/java/transact/logic/commands/AddTransactionCommand.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import transact.logic.commands.exceptions.CommandException;
 import transact.model.Model;
 import transact.model.transaction.Transaction;
+import transact.ui.MainWindow;
```
Comment (cxo05) [inline src/main/java/transact/logic/commands/AddTransactionCommand.java:16]:
```suggestion
import transact.ui.MainWindow.TabWindow;
```

**Review by cxo05 — CHANGES_REQUESTED** [review]

File: src/main/java/transact/logic/commands/AddTransactionCommand.java (hunk 1 above)
Comment (cxo05) [inline src/main/java/transact/logic/commands/AddTransactionCommand.java:61]:
```suggestion
        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)), TabWindow.TRANSACTIONS);
```

File: src/main/java/transact/logic/commands/AddTransactionCommand.java (hunk 2 above)
Comment (Bombbird2001) [reply src/main/java/transact/logic/commands/AddTransactionCommand.java:16]: Updated

File: src/main/java/transact/logic/commands/AddTransactionCommand.java (hunk 1 above)
Comment (Bombbird2001) [reply src/main/java/transact/logic/commands/AddTransactionCommand.java:61]: Updated

**Comment by cxo05** [comment]
Ok LGTM

**Comment by cxo05** [comment]
@AlainS87 @PetrichorPrecipice Can take a look too and give comment

**Review by cxo05 — APPROVED** [review]

**Comment by AlainS87** [comment]
Looks good

**Comment by PetrichorPrecipice** [comment]
LGTM

**Pull request merged**